### PR TITLE
Added script for problem 1, 2, 3 and 4

### PR DESCRIPTION
**Pull request by 0xbhavik**
Script for prob_1a, prob_1b, prob_2, prob_3 and prob_4.

### CODE REVIEW

File: week_1/.prob3.sh.swp
Comment (subham-padhi-01) [inline week_1/.prob3.sh.swp:1]: is this swap file required ?

Comment (0xbhavik) [reply week_1/.prob3.sh.swp:1]: No. it got created since i was using multiple instances of vim. it got pushed accidently

File: week_1/prob_1b.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#! /bin/bash
+
+cat web-server.log | awk -F " " '{print $9}' | sort | uniq -c | awk '{print $2 ": " $1}'
```
Comment (prince-inito) [inline week_1/prob_1b.sh]: can you print it's output in a file and push that file?

Comment (0xbhavik) [reply week_1/prob_1b.sh]: Done

File: week_1/output_prob_1b.txt
Comment (0xbhavik) [inline week_1/output_prob_1b.txt:1]: @prince-inito

File: week_1/prob_4.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+echo "enter remote server's ip address"
+read remoteIpAddr
+
+
+for i in $(seq 0 $((len-1))); 
+do
+
+ipAddr=${ipAddrs[$i]}
+userName=${userNames[$i]}
+publicKey=${publicKeys[$i]}
```
Comment (prince-inito) [inline week_1/prob_4.sh:25]: from where you are getting this public key? As discussed you should get path of public file and then you have to check it

File: week_1/prob3.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ then hasLowerCase=1
+ elif [[ $char == [1-9] ]];
+ then hasNumber=1
+ else 
+      hasSpecialChar=1
+ fi 
+done
+
+
+if [[ str_len -ge "10" ]]  && [[ hasUpperCase -eq 1 ]] && [[ hasLowerCase -eq 1 ]] && [[ hasSpecialChar -eq 1 ]];
```
Comment (subham-padhi-01) [inline week_1/prob3.sh:26]: Probably we can think of a better logic to check password strength.

File: week_1/prob_2_monitor.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+echo "process stopped, starting it again"
+bash $1 &
+pid=$!
+
+else 
+echo "process is already running"
+
+fi
+
+sleep 3
```
Comment (subham-padhi-01) [inline week_1/prob_2_monitor.sh:23]: why is the sleep required ?

File: week_1/prob_2_process.sh
Comment (subham-padhi-01) [inline week_1/prob_2_process.sh:1]: this is not required in the PR, please remove

**Pull request merged**